python/ray/exceptions.py

```diff
@@ -292,19 +295,103 @@ def __init__(self, object_ref_hex, call_site):
             ray_constants.CALL_STACK_LINE_DELIMITER, "\n  ")
 
     def __str__(self):
-        msg = (f"Object {self.object_ref_hex} cannot be retrieved due to node "
-               "failure or system error.")
+        msg = f"Object {self.object_ref_hex} cannot be retrieved. "
         if self.call_site:
-            msg += (f" The ObjectRef was created at: {self.call_site}")
+            msg += (f"The ObjectRef was created at: {self.call_site}")
         else:
             msg += (
-                " To see information about where this ObjectRef was created "
+                "To see information about where this ObjectRef was created "
                 "in Python, set the environment variable "
                 "RAY_record_ref_creation_sites=1 during `ray start` and "
                 "`ray.init()`.")
         return msg
 
 
+class ObjectLostError(ObjectUnreachableError):
+    """Indicates that the object is lost from distributed memory, due to
+    node failure or system error.
+
+    Attributes:
+        object_ref_hex: Hex ID of the object.
+    """
+
+    def __str__(self):
+        return super().__str__() + "\n\n" + (
+            f"All copies of {self.object_ref_hex} are lost "
+            "due to node failure.\n\n"
+            "If you did not receive a message about a worker node "
+            "dying, this is likely a system-level bug. "
+            "Please file an issue on GitHub at "
```

Comment:
Prefer to remove "file an issue" logs. I think if I am a newbie user, I'd just think it is that Ray is unstable (since the message mentions the system level bug).